site/app/controllers/admin/ConfigurationController.php

```diff
@@ -71,55 +71,42 @@ public function viewConfiguration() {
 
     public function updateConfiguration() {
         if (!$this->core->checkCsrfToken($_POST['csrf_token'])) {
-            $this->core->addErrorMessage("Invalid CSRF token. Try again.");
-            $_SESSION['request'] = $_POST;
-            $this->core->redirect($this->core->buildUrl(array('component' => 'admin',
-                                                              'page' => 'configuration',
-                                                              'action' => 'view')));
+            $this->core->getOutput()->renderJsonFail('Invalid CSRF token');
```

Comment:
I think you want to `return` in this case to stop control flow.  On a similar note, the only reason some controller methods return a value is so the PHPUnit tests can get the result of the controller without mocking the view.